### ISSUE TITLE
[Persistence_representations - Persistence_heat_maps] Simplify distance_from_diagonal_scaling

### ISSUE THREAD

**Issue by VincentRouvreau** [GUDHI/gudhi-devel#19]
`distance_from_diagonal_scaling` and `squared_distance_from_diagonal_scaling` have complicated implementations with pow, sqrt, etc (all missing `std::`). As far as I can tell, up to a multiplicative constant (which we can probably omit), the first one should be just `std::abs(p.second-p.first)` and the second one the square of `p.second-p.first`.

**Comment by yagu0** [comment]
Indeed. Moreover the comments are confusing: if the argument is the point coordinates, isn't p.second - p.first the lifespan of a point on the diagram ? So it wouldn't correspond to the "Euclidean distance of (b,d) from diagonal".

**Comment by mglisse** [comment]
It actually computes the projection on the diagonal `(b+d)/2`, and then the squared distance to that. It happens to be equal to `|d-b|/sqrt(2)`. When computing the distance to the diagonal, Manhattan, Chebyshev and Euclidean are all proportional.
